pkg/plugin/settings.go

```diff
@@ -10,8 +10,10 @@ import (
 const openAIKey = "openAIKey"
 
 type OpenAISettings struct {
-	URL            string `json:"url"`
-	OrganizationID string `json:"organizationId"`
+	URL            string            `json:"url"`
+	OrganizationID string            `json:"organizationId"`
+	UseAzure       bool              `json:"azureOpenAI"`
```

Comment:
It might be nicer to use an open type here rather than a `bool`. Something like

```go
type openAIType string

const (
	openAITypeOpenAI openAIType "openai"
	openAITypeAzure  openAIType "azure"
)
```

above, then

```suggestion
	Type       openAIType            `json:"type"`
```

in the settings.

(in case in future people want support for other providers which look and act like OpenAI)